src/mca/gds/ds12/gds_ds12_base.c

```diff
@@ -23,15 +23,22 @@
 #include "src/mca/common/dstore/dstore_common.h"
 #include "gds_ds12_base.h"
 #include "gds_ds12_lock.h"
+#include "gds_ds12_file.h"
 #include "src/mca/common/dstore/dstore_base.h"
 
 static pmix_common_dstore_ctx_t *ds12_ctx;
 
 static pmix_status_t ds12_init(pmix_info_t info[], size_t ninfo)
 {
     pmix_status_t rc = PMIX_SUCCESS;
+    pmix_common_dstore_file_cbs_t *dstore_file_cbs = NULL;
 
-    ds12_ctx = pmix_common_dstor_init(&pmix_ds12_lock_module, "ds12", info, ninfo);
+    if (!PMIX_PROC_IS_SERVER(pmix_globals.mypeer)) {
+        dstore_file_cbs = &pmix_ds20_file_module;
```

Comment:
Why no ds12 here?

Reply:
The ds12 module has two dstore formats for 12 and 20. By default, it uses v20 format for dstor, but if a client will v12 then it is changed to v12.